pkg/cloudclient/aws/private.go

```diff
@@ -217,26 +220,45 @@ func generateUserData() (string, error) {
 func (c Client) findUnreachableEndpoints(ctx context.Context, instanceID string) ([]string, error) {
 	var match []string
 
+	// Compile the regular expressions once
+	reVerify := regexp.MustCompile(userdataEndVerifier)
+	reUnreachableErrors := regexp.MustCompile(`Unable to reach (\S+)`)
+
+	latest := true
+	input := ec2.GetConsoleOutputInput{
+		InstanceId: &instanceID,
+		Latest:     &latest,
+	}
+
 	err := wait.PollImmediate(30*time.Second, 10*time.Minute, func() (bool, error) {
-		output, err := c.ec2Client.GetConsoleOutput(ctx, &ec2.GetConsoleOutputInput{InstanceId: &instanceID})
+		output, err := c.ec2Client.GetConsoleOutput(ctx, &input)
 		if err == nil && output.Output != nil {
-			// Find unreachable targets from output
+			// First, gather the ec2 console output
 			scriptOutput, err := base64.StdEncoding.DecodeString(*output.Output)
 			if err != nil {
 				// unable to decode output. we will try again
+				c.logger.Debug(ctx, "Error while collecting console output, will retry on next check interval: %s", err)
 				return false, nil
 			}
 
-			// If debug logging is enabled, output the full console log
-			c.logger.Debug(ctx, "Full EC2 console output:\n---\n%s\n---", scriptOutput)
+			// In the early stages, an ec2 instance may be running but the console is not populated with any data, retry if that is the case
+			if len(scriptOutput) < 1 {
+				c.logger.Debug(ctx, "EC2 console output not yet populated with data, continuing to wait...")
+				return false, nil
+			}
 
-			if !strings.Contains(string(scriptOutput), "USERDATA END") {
```

Comment:
Replaced this check @bng0y added with a regexp and ensured we use a single variable to define the string we are looking for, used in both places we rely on it so if it ever changes, both are updated.